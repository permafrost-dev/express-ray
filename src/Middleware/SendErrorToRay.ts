import { ray } from 'node-ray';

export const sendErrorToRay = (err, req, res, next, rayInstance: any = null) => {
    const { method, url, params, query } = req;

    rayInstance = rayInstance ?? ray();

    rayInstance.table(
        {
            error: err.message,
            method,
            url,
            params,
            query,
        },
        'Error',
    );

    if (res.headersSent) {
        return next(err);
    }

    next(err);
};

export default sendErrorToRay;
