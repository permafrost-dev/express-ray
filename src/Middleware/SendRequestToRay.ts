import { ray } from 'node-ray';

export const sendRequestToRay = (req, res, next, rayInstance: any = null) => {
    const { method, url, params, query, headers } = req;

    rayInstance = rayInstance ?? ray();

    const data = {
        method,
        url,
        params,
        query,
        headers,
    };

    if (method.toLowerCase() === 'post') {
        data['body'] = req.body;
    }

    rayInstance.table(data, 'Request');

    next();
};

export default sendRequestToRay;
