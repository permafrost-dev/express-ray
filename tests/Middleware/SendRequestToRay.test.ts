/* eslint-disable @typescript-eslint/no-empty-function */

import { sendRequestToRay } from '@/Middleware/SendRequestToRay';
import { FakeRay } from '@tests/TestClasses/FakeRay';

it('sends a get request to ray', () => {
    const rayInstance = new FakeRay();

    const request = {
        method: 'GET',
        url: '/',
        path: '/',
        params: {},
        query: {},
    };

    sendRequestToRay({})(request, {}, () => {}, rayInstance);

    expect(rayInstance.sentPayloads).toHaveLength(1);
    expect(rayInstance.sentPayloads).toMatchSnapshot();
});

it('sends a post request to ray', () => {
    const rayInstance = new FakeRay();

    const request = {
        method: 'POST',
        url: '/',
        path: '/',
        params: {},
        query: {},
        body: { message: 'test' },
    };

    sendRequestToRay({})(request, {}, () => {}, rayInstance);

    expect(rayInstance.sentPayloads).toHaveLength(1);
    expect(rayInstance.sentPayloads).toMatchSnapshot();
});
