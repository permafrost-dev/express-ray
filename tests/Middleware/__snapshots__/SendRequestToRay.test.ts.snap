// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sends a get request to ray 1`] = `
Array [
  Object {
    "data": Object {
      "headers": undefined,
      "method": "GET",
      "params": Object {},
      "query": Object {},
      "url": "/",
    },
    "label": "Request",
  },
]
`;

exports[`sends a post request to ray 1`] = `
Array [
  Object {
    "data": Object {
      "body": Object {
        "message": "test",
      },
      "headers": undefined,
      "method": "POST",
      "params": Object {},
      "query": Object {},
      "url": "/",
    },
    "label": "Request",
  },
]
`;
